updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -69,16 +69,14 @@ def groups
       @dependency_group_engine.dependency_groups
     end
 
-    def calculate_ungrouped_dependencies(dependencies_handled)
-      @ungrouped_dependencies = allowed_dependencies.reject { |dep| dependencies_handled.include?(dep.name) }
-    end
-
     def ungrouped_dependencies
       # If no groups are defined, all dependencies are ungrouped by default.
       return allowed_dependencies unless groups.any?
-      return @ungrouped_dependencies if defined?(@ungrouped_dependencies)
 
-      @dependency_group_engine.ungrouped_dependencies
+      # Otherwise return dependencies that haven't been handled during the group update portion.
+      all_handled_dependencies = Set.new
+      groups.each { |group| all_handled_dependencies += group.handled_dependencies }
+      allowed_dependencies.reject { |dep| all_handled_dependencies.include?(dep.name) }
```

Comment:
Would this work?

```suggestion
      allowed_dependencies.reject { |dep| groups.any? { |group| group.handled_dependencies.include?(dep.name) } }
```

Reply:
I wrote it this way first pass, but it seemed a bit unclear. I liked being able to name `all_handled_dependencies`. Now that I'm looking at it again, I kind of like this instead:

```ruby
      all_handled_dependencies = Set.new(groups.flat_map(&:handled_dependencies))
      allowed_dependencies.reject { |dep| all_handled_dependencies.include?(dep.name) }
```

I'm trying not to code golf it, but also keep it brief. Also the Set prevents any kind of silly `O(n^2)` for our monorepo friends.

Reply:
Yep, makes sense 👍